runtime/common/src/mmr.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A pallet responsible for creating Merkle Mountain Range (MMR) leaf for current block.
+
+use beefy_primitives::ValidatorSetId;
+use sp_core::H256;
+use sp_runtime::traits::Convert;
+use sp_std::prelude::*;
+use frame_support::{decl_module, decl_storage, RuntimeDebug};
+use pallet_mmr::primitives::LeafDataProvider;
+use parity_scale_codec::{Encode, Decode};
+use runtime_parachains::paras;
+
+/// A BEEFY consensus digest item with MMR root hash.
+pub struct DepositBeefyDigest<T>(sp_std::marker::PhantomData<T>);
+
+impl<T> pallet_mmr::primitives::OnNewRoot<beefy_primitives::MmrRootHash> for DepositBeefyDigest<T> where
+	T: pallet_mmr::Config<Hash = beefy_primitives::MmrRootHash>,
+	T: pallet_beefy::Config,
+{
+	fn on_new_root(root: &<T as pallet_mmr::Config>::Hash) {
+		let digest = sp_runtime::generic::DigestItem::Consensus(
+			beefy_primitives::BEEFY_ENGINE_ID,
+			parity_scale_codec::Encode::encode(
+				&beefy_primitives::ConsensusLog::<<T as pallet_beefy::Config>::AuthorityId>::MmrRoot(*root)
+			),
+		);
+		<frame_system::Module<T>>::deposit_log(digest);
+	}
+}
+
+/// Convert BEEFY secp256k1 public keys into uncompressed for
+pub struct UncompressBeefyEcdsaKeys;
```

Comment:
So there is no extra information (hence it's called compression :)) it's just smaller in size because of different, but equivalent representation of the curve points. The issue is that Ethereum requires uncompressed keys, and decompression in Solidity might be quite inefficient cause it requires a bunch of curve operations (and there is no built-in for that). Hence to optimize gas costs of Ethereum verification (which is the primary consumer of this API) we use decompress in Substrate